src/System/include/BipedalLocomotion/System/VariablesHandler.h

```diff
@@ -0,0 +1,56 @@
+/**
+ * @file VariablesHandler.h
+ * @authors Giulio Romualdi
+ * @copyright 2020 Istituto Italiano di Tecnologia (IIT). This software may be modified and
+ * distributed under the terms of the GNU Lesser General Public License v2.1 or any later version.
+ */
+
+#include <string>
+#include <unordered_map>
+
+#include <iDynTree/Core/Utils.h>
+
+#ifndef BIPEDAL_LOCOMOTION_SYSTEM_ADVANCEABLE_VARIABLES_HANDLER_H
+#define BIPEDAL_LOCOMOTION_SYSTEM_ADVANCEABLE_VARIABLES_HANDLER_H
+
+namespace BipedalLocomotion
+{
+namespace System
+{
+/**
+ * VariableHandler is useful to handle variables in an optimization problem, Their name, dimension
+ * and position
+ */
+class VariablesHandler
+{
+
+    std::unordered_map<std::string, iDynTree::IndexRange> m_variables; /**< Map containing the name
```

Comment:
Since you are using an IndexRange, I guess that the assumption is to store all the values in a vector. Does it make sense to add a container to store the values in this class? In this way, once you fill the whole buffer (maybe from the solution of the optimization variable), you can get the values directly using ``getVariable``, maybe using a GenericVector as a output. In fact, you can use ``span`` to get a reference to a sub-vector, then you can use it to build a GenericVector.

Reply:
After a short Fm2mF (Face-mask to mask-face) discussion we decided to avoid implementing this in the current PR. Since we are thinking to introduce `CasADi` in our pipeline, we may need different implementations of the `handler`.